pkg/cloudprovider/openstack/openstack.go

```diff
@@ -630,31 +631,49 @@ func (lb *LoadBalancer) UpdateTCPLoadBalancer(name, region string, hosts []strin
 	return nil
 }
 
-func (lb *LoadBalancer) DeleteTCPLoadBalancer(name, region string) error {
-	glog.V(4).Infof("DeleteTCPLoadBalancer(%v, %v)", name, region)
-
-	vip, err := getVipByName(lb.network, name)
-	if err != nil {
-		return err
-	}
-
-	pool, err := pools.Get(lb.network, vip.PoolID).Extract()
-	if err != nil {
-		return err
+func (lb *LoadBalancer) EnsureTCPLoadBalancerDeleted(name, region string) error {
+	glog.V(4).Infof("EnsureTCPLoadBalancerDeleted(%v, %v)", name, region)
+
+	// TODO: Because the we look up the pool using the VIP object, if the VIP is
+	// already gone we can't attempt to delete the pool. We should instead look up
+	// the pool by name and attempt to delete it even if the VIP has already been
+	// deleted.
+	vip, vipErr := getVipByName(lb.network, name)
+	if vipErr == ErrNotFound {
+		return nil
+	} else if vipErr != nil {
+		return vipErr
+	}
+	vipExists := (vipErr == nil)
```

Comment:
I don't think that it's possible to get here without vipErr being nil, in which case vipExists is superfluous, right?